src/components/forms/shipment/ShipmentOrderData.tsx

```diff
@@ -0,0 +1,122 @@
+
+  const toggleAddModal = () => {
+    setDisplayAddModal(!displayAddModal);
+  };
+
+  const toggleShowOrderModal = (object?: FieldValues) => {
+    if (object) {
+      const lineItems = orderDetails.filter(
+        (order: any) => order.order_details_id === object.order_details_id,
+      ).map((order: any) => order.line_items);
+      setOrderLineItemToShow(lineItems);
+    }
+    setDisplayShowOrderModal(!displayShowOrderModal);
+  };
+
+  const onSubmitAdd = (formValues: FieldValues) => {
+    onOrderDetailAdd(formValues);
+    setOrderDetailList(orderDetails);
+    toggleAddModal();
+  };
+
+  const onDelete = (object: FieldValues) => {
+    onOrderDetailDelete(object);
+    const arrayDeleted = orderDetails.filter(
+      (order: any) => order.order_details_id !== object.order_details_id,
+    );
+    setOrderDetailList(arrayDeleted);
+  };
+
+  const columns: ColumnType[] = React.useMemo(() => [
+    {
+      Header: format('shipment.customer_name.label'),
+      accessor: 'customer_name',
+    },
+    {
+      Header: format('shipment.order_number.label'),
+      accessor: 'order_details_id',
+    },
+  ], [format]);
+
+  const editInitialValue = () => orderDetails.map(
+    (object: any) => ({ id: object.customer, name: object.customer_name }),
+  )[0];
+
+  return (
+    <>
+      <div className="flow-root">
+        <p className="float-left text-[21px] text-transit-black font-semibold">{format('shipment.orders.label')}</p>
+        <SubmitButton onClick={toggleAddModal} className="float-right h-fit w-fit" title={format('shipment.order_add.label')} />
+      </div>
+      <div>
+        <TableWithoutPagination
+          columns={columns}
+          data={orderDetailList.map((row: any) => (
+            { customer_name: row.customer_name, order_details_id: row.order_details_id }
+          ))}
+          deleteAction={onDelete}
+          editAction={toggleShowOrderModal}
+        >
+          <div />
+        </TableWithoutPagination>
+      </div>
+      <Dialog
+        isOpen={displayAddModal}
+        onClose={toggleAddModal}
+        setCustomDialogContent
+        // eslint-disable-next-line
+        children={[
+          <ShipmentOrderDetailsForm
+            onSubmit={onSubmitAdd}
+            onCancel={toggleAddModal}
+            title={`${format('shipment.add_order_to_shipment.label')}`}
+            initialFormValue={mode === 'Add' || (mode === 'Edit' && orderDetails.length === 0) ? {} : { customer: editInitialValue() }}
+            mode="Add"
+            submitButtonText={format('app.add')}
+            currentData={orderDetailList.map((row: any) => (row.order_details_id))}
+          />,
+        ]}
+      />
+      <Dialog
+        isOpen={displayShowOrderModal}
+        onClose={toggleShowOrderModal}
+        setCustomDialogContent
+        // eslint-disable-next-line
+        children={[
+          <ShipmentOrderDetailsForm
+            onSubmit={() => {}}
```

Comment:
Is it ok here to have empty function? Or it's mistake? Or just remove those props

Reply:
yes, this is for purpose to do nothing. This anyway will be changed while working on photos